common/src/test/java/org/apache/sedona/common/raster/RasterBandAccessorsTest.java

```diff
@@ -84,6 +84,41 @@ public void testBandNoDataValueIllegalBand() throws FactoryException, IOExceptio
     assertEquals("Provided band index 2 is not present in the raster", exception.getMessage());
   }
 
+  @Test
```

Comment:
Please add a test to show how this works in a real Sedona RS function. I am not sure if it can successfully convert Double.NaN to Spark null.